cpp/arcticdb/version/version_core.cpp

```diff
@@ -304,92 +324,137 @@ void check_update_data_is_sorted(InputTensorFrame& frame, const index::IndexSegm
         "When calling update, the existing data must be sorted.");
 }
 
-VersionedItem update_impl(
+struct UpdateRanges {
+    IndexRange front;
+    IndexRange back;
+    IndexRange original;
+};
+
+folly::Future<AtomKey> async_update_impl(
```

Comment:
This methods a bit of a monster now, if there's a clean way to break it up a bit it should get more readable

Reply:
I couldn't think way to split things in logically rather than pulling them for the sake of splitting. I'll give it another try, now that everything compiles and all tests are passing there will be less unknowns.